src/System.IO.FileSystem/tests/Directory/GetFileSystemEntries_str_str.cs

```diff
@@ -639,6 +639,13 @@ public void SearchPatternLongPath()
 
             if (TestFiles)
             {
+                if (PlatformDetection.IsNetNative)
+                {
+                    // UapAot throws if the Directory of the file to create doesn't exist.
+                    Assert.Throws<DirectoryNotFoundException>(() => File.Create(longFullname));
```

Comment:
If it's exhibiting different behavior, isn't that a bug we should fix in the implementation?